pictorus-internal/src/loggers/csv_logger.rs

```diff
@@ -220,4 +343,204 @@ mod tests {
+            vector: Some([[99.99, 88.88, 77.77]]),
+            scalar: None,
+            matrix: None,
+            bytesarray: None,
+        };
+
+        // Second log with short string
+        let log_data2 = TestLogData {
+            state_id: Some("short".to_string()),
+            timestamp: Some(2.0),
+            utctime: None,
+            vector: None,
+            scalar: Some(1.0),
+            matrix: None,
+            bytesarray: None,
+        };
+
+        logger.log(&log_data1, Duration::ZERO);
+        logger.log(&log_data2, Duration::from_millis(100));
+
+        drop(logger);
+
+        let contents = std::fs::read_to_string(output_path).unwrap();
+        let lines: Vec<&str> = contents.lines().collect();
+
+        // Verify second line doesn't contain remnants of first
+        assert!(!lines[2].contains("very_long"));
+        assert!(!lines[2].contains("99.99"));
+        assert!(lines[2].contains("\"short\""));
+    }
+
+    #[test]
+    fn test_flush_on_drop() {
+        let temp_dir = tempfile::TempDir::new().unwrap();
+        let output_path = temp_dir.path().join("test_flush.csv");
+
+        let log_data = TestLogData {
+            state_id: Some("flush_test".to_string()),
+            timestamp: Some(123.456),
+            utctime: None,
+            vector: None,
+            scalar: Some(789.0),
+            matrix: None,
+            bytesarray: None,
+        };
+
+        {
+            let mut logger = CsvLogger::new(Duration::from_micros(1), output_path.clone());
+            logger.log(&log_data, Duration::ZERO);
+            // Logger dropped here - should flush
+        }
+
+        // Verify data was written
+        let contents = std::fs::read_to_string(output_path).unwrap();
+        assert!(contents.contains("flush_test"));
+        assert!(contents.contains("123.456"));
+        assert!(contents.contains("789"));
+    }
+
+    #[test]
+    fn test_large_array_handling() {
+        let temp_dir = tempfile::TempDir::new().unwrap();
+        let output_path = temp_dir.path().join("test_large.csv");
+
+        // Create a struct with a large array
+        #[derive(serde::Serialize)]
+        struct LargeArrayData {
+            #[serde(rename = "large_array")]
+            large_array: Option<Vec<f64>>,
+            id: Option<String>,
+        }
+
+        let large_data = LargeArrayData {
+            large_array: Some((0..100).map(|i| i as f64 * 0.1).collect()),
+            id: Some("large_test".to_string()),
+        };
+
+        let mut logger = CsvLogger::new(Duration::from_micros(1), output_path.clone());
+
+        let start = std::time::Instant::now();
+        logger.log(&large_data, Duration::ZERO);
+        let elapsed = start.elapsed();
+
+        drop(logger);
+
+        // Should complete quickly even with large array
+        assert!(
+            elapsed < Duration::from_millis(10),
+            "Large array logging took too long: {elapsed:?}"
+        );
```

Comment:
I'm not sure this belongs in a unit test. These kind of perf comparisons are highly dependent on the platform and state of the system. It could be worth setting up some kind of benchmark suite for this, but even then we would want to make sure it's run in a semi-controlled environment